express/blocks/template-list/template-list.js

```diff
@@ -1859,36 +1877,64 @@ function addBackgroundAnimation($block, animationUrl) {
   }
 }
 
-export default async function decorate($block) {
-  if ($block.classList.contains('spreadsheet-powered')) {
-    const placeholders = await fetchPlaceholders().then((result) => result);
-    const relevantRowsData = await fetchRelevantRows(window.location.pathname);
-    props.limit = parseInt(placeholders['relevant-rows-templates-limit'], 10) || 10;
-
-    if (relevantRowsData) {
-      $block.closest('.section').dataset.audience = 'mobile';
-
-      props.headingTitle = relevantRowsData.header || null;
-      props.headingSlug = relevantRowsData.shortTitle || null;
-      props.viewAllLink = relevantRowsData.viewAllLink || null;
-
-      $block.innerHTML = $block.innerHTML.replaceAll('default-title', relevantRowsData.shortTitle || '');
-      $block.innerHTML = $block.innerHTML.replaceAll('default-tasks', relevantRowsData.templateTasks || '');
-      $block.innerHTML = $block.innerHTML.replaceAll('default-topics', relevantRowsData.templateTopics || '');
-      $block.innerHTML = $block.innerHTML.replaceAll('default-locale', relevantRowsData.templateLocale || 'en');
-      $block.innerHTML = $block.innerHTML.replaceAll('default-premium', relevantRowsData.templatePremium || '');
-      $block.innerHTML = $block.innerHTML.replaceAll('default-animated', relevantRowsData.templateAnimated || '');
-      $block.innerHTML = $block.innerHTML.replaceAll('https://www.adobe.com/express/templates/default-create-link', relevantRowsData.createLink || '/');
-      $block.innerHTML = $block.innerHTML.replaceAll('default-format', relevantRowsData.placeholderFormat || '');
-
-      if (relevantRowsData.templateTasks === '') {
-        $block.innerHTML = $block.innerHTML.replaceAll('default-create-link-text', placeholders['start-from-scratch'] || '');
-      } else {
-        $block.innerHTML = $block.innerHTML.replaceAll('default-create-link-text', relevantRowsData.createText || '');
+async function replaceRRTemplateList($block) {
+  const placeholders = await fetchPlaceholders().then((result) => result);
+  const relevantRowsData = await fetchRelevantRows(window.location.pathname);
+  props.limit = parseInt(placeholders['relevant-rows-templates-limit'], 10) || 10;
+
+  if (relevantRowsData) {
+    $block.closest('.section').dataset.audience = 'mobile';
```

Comment:
Blocks shouldn't really be modifying stuff in the DOM outside of themselves and the block context.
Can we take a look at this together because I don't understand enough of what's going on to propose a different approach. :)

Reply:
Yes let's find time to look at this together if you don't mind!

Reply:
perfect !